src/dbus_api.rs

```diff
@@ -355,7 +372,60 @@ fn set_filesystem_mountpoint(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult
 }
 
 fn rename_filesystem(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
-    Ok(vec![m.msg.method_return().append3("/dbus/cache/path", 0, "Ok")])
+    let message: &Message = m.msg;
+    let mut iter = message.iter_init();
+
+    let new_name: &str = try!(get_next_arg(&mut iter, 0));
+
+    let dbus_context = m.path.get_data();
+    let object_path = m.path.get_name();
+    let return_message = message.method_return();
+    let return_sig = "qs";
+    let default_return = MessageItem::Array(vec![], return_sig.into());
+
+    let (pool_name, filesystem_name) =
+        dbus_try!(object_path_to_pair(dbus_context, object_path); default_return; return_message);
+
+    let mut b_engine = dbus_context.engine.borrow_mut();
+    let ref mut pool = engine_try!(b_engine.get_pool(&pool_name);
+                                   default_return;
+                                   return_message);
+
```

Comment:
Added check to make sure new_name doesn't already exist.  Will return error now if new_name is found.